build-system/tasks/check-sourcemaps.js

```diff
@@ -75,7 +75,11 @@ function checkSourcemapUrl(sourcemapJson, map) {
     throw new Error('Could not find sourcemap URL');
   }
   if (!sourcemapJson.sourceRoot.match(sourcemapUrlMatcher)) {
-    log(red('ERROR:'), cyan(sourcemapJson.sourceRoot), 'is badly formatted');
+    log(
+      red('ERROR:'),
+      cyan(/** @type {string} */ (sourcemapJson.sourceRoot)),
```

Comment:
So for the instances of `cyan(/** @type {string} */ (x))` - it looks busy and not very helpful. Is there a way we can suppress these kind of errors instead?

Reply:
Built-in type info for the text coloring library we use can be found at https://github.com/lukeed/kleur/blob/master/index.d.ts. Perhaps we can override it to take anything that can be logged to the console (including things like `null` and `undefined`)?

Reply:
I'm unaware of any ways to override the signature (I would be super interested if anyone does) but I imagine this could be accomplished by importing all of `kleur` into a file then exporting each of its members with a different signature. Unfortunately that would require changing a huge number of files in `build-system` but would give us greater control in the end. Given the scope of the change I'd love some feedback prior to making it.
Should I just remove these sorts of type assertions from this PR and handle `kleur` problems all at once?

Reply:
Punting this to a separate PR SGTM. Will do some homework on how to silence the log-color-specific type errors.

Reply:
Cool, thanks Raghu and Riley.

Reply:
> Unfortunately that would require changing a huge number of files in build-system but would give us greater control in the end. Given the scope of the change I'd love some feedback prior to making it. Should I just remove these sorts of type assertions from this PR and handle kleur problems all at once?

Did some homework. There are 75 files that use `kleur/colors` in `build-system/`, 1 in the repo root, and 1 in `validator/` (this one can be ignored). The idea you describe above SGTM. We could create `build-system/common/colors.js` (similar to `logging.js`) and wrap / export [these functions](https://github.com/lukeed/kleur/blob/598f24cb7f5054b7c504e9ab89251305311131b1/colors.mjs#L23-L53) exported by `kleur/colors` such that they can be used with any type.

Reply:
#34243 is merged so I'm resolving this.